src/examples/mc_pos_control_multiplatform/mc_pos_control.cpp

```diff
@@ -930,12 +932,14 @@ void  MulticopterPositionControlMultiplatform::handle_vehicle_attitude(const px4
 						_R(i, 2) = body_z(i);
 					}
 
-					/* copy rotation matrix to attitude setpoint topic */
-					memcpy(&_att_sp_msg.data().R_body[0], _R.data, sizeof(_att_sp_msg.data().R_body));
-					_att_sp_msg.data().R_valid = true;
-
 					/* calculate euler angles, for logging only, must not be used for control */
 					math::Vector<3> eul = _R.to_euler();
+
+					math::Quaternion q;
+					q.from_dcm(_R);
+					memcpy(_att_sp_msg.data().q_d, q.data, sizeof(_att_sp_msg.data().q_d));
+					_att_sp_msg.data().q_d_valid = true;
```

Comment:
move block above "calculate euler angles...", such that computation and assignment are together